analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Nested()
+    {
+        /*       Block 1
+         *       #0=s1
+         *     #0 is null
+         *        /   \
+         *      F/     \T
+         *      /       \
+         *   Block 2  Block 3---+------F----> Block 4 --------+
+         *   #1=#0   s2 is null |             #2=s3           |
+         *       \     /        |                             |
+         *        Block 6       +------T----> Block 5-------->|
+         *         s=#1                   #2="NestedFalse"    |
+         *          |                                         |
+         *          |                                Block6<--+
+         *          |                              #2 is null
+         *          |                               /   \
+         *          |                             T/     \F
+         *          |                             /       \
+         *          |                           Block8   Block7
+         *          |                         #1="Hello"  #1=#2"
+         *          |                             |         |
+         *          +------------>Block9<---------+---------+
+         *                         s=#1
+         *                          |
+         *                         Exit
+         */
+        const string code = """string s = s1 ?? (s2 is null ? s3 : "NestedFalse") ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1, string s2, string s3");
+        context.ValidateEntry(LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s3"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Overwrite()
+    {
+        /*       Block 1
+         *       #0=s1
+         *         |
+         *       Block 2
+         *   #1=(s1="overwrite")
+         *     #1 is null
+         *        /   \
+         *       T     F
```

Comment:
You have them swapped. There's a `WhenTrue` semantics to confuse the enemy
```suggestion
         *       F     T
```